framework/helpers/BaseHtml.php

```diff
@@ -1719,7 +1719,8 @@ public static function renderSelectOptions($selection, $items, &$tagOptions = []
         );
         $options['encode'] = ArrayHelper::getValue($options, 'encode', $encode);
 
-        if (null !== ($prompt = ArrayHelper::remove($tagOptions, 'prompt'))) {
+        $prompt = ArrayHelper::remove($tagOptions, 'prompt');
+        if (null !== $prompt) {
```

Comment:
well, have you read the 'Criticism' part on Wikipedia?:
'Critics of Yoda conditions see the lack of readability as a disadvantage that outweighs the benefits described....'